inference-engine/src/gna_plugin/transformations/convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,547 @@
+    size_t filter_dilation_width;
+    size_t filter_dilation_height;
+    size_t filter_stride_width;
+    size_t filter_stride_height;
+    size_t pads_begin_width;
+    size_t pads_begin_height;
+    size_t pads_end_width;
+    size_t pads_end_height;
+    ngraph::op::PadType padding_type;
+    ngraph::Shape output_shape;
+    ngraph::element::Type element_type;
+};
+
+static bool VerifyAndGetConvParams(std::shared_ptr<ngraph::opset7::Convolution> conv, ConvData& conv_data) {
+    const auto& input = conv->input_value(0);
+
+    // We support only 2D conv batch 1
+    if (conv->get_dilations().size() != 2 ||
+        conv->get_strides().size() != 2 ||
+        input.get_shape()[0] != 1) {
+        return false;
+    }
+
+    conv_data.output_shape = conv->get_output_shape(0);
+    conv_data.padding_type = conv->get_auto_pad();
+    conv_data.input_channel_count = conv->input_value(0).get_shape()[1];
+    conv_data.input_height = conv->input_value(0).get_shape()[2];
+    conv_data.input_width = conv->input_value(0).get_shape()[3];
+    conv_data.filter_count = conv->input_value(1).get_shape()[0];
+    conv_data.filter_height = conv->input_value(1).get_shape()[2];
+    conv_data.filter_width = conv->input_value(1).get_shape()[3];
+    conv_data.filter_dilation_height = conv->get_dilations()[0];
+    conv_data.filter_dilation_width = conv->get_dilations()[1];
+    conv_data.filter_stride_height = conv->get_strides()[0];
+    conv_data.filter_stride_width = conv->get_strides()[1];
+    conv_data.pads_begin_height = conv->get_pads_begin()[0];
+    conv_data.pads_begin_width = conv->get_pads_begin()[1];
+    conv_data.pads_end_height = conv->get_pads_end()[0];
+    conv_data.pads_end_width = conv->get_pads_end()[1];
+    conv_data.element_type = conv->get_element_type();
+
+    IE_ASSERT(conv_data.filter_count == conv_data.output_shape[1]);
+
+    return true;
+}
+
+static bool TransposeOrderMatches(std::shared_ptr<ngraph::opset7::Transpose> transpose, std::vector<int64_t> order) {
+    if (!transpose)
+        return false;
+    const ngraph::Output<ngraph::Node>& transpose_order = transpose->input_value(1);
+    auto transpose_order_dim = transpose_order.get_shape().size();
+
+    if (transpose_order_dim != 1 || transpose_order.get_shape()[0] != order.size())
+        return false;
+
+    auto const_with_order_values = std::dynamic_pointer_cast<ngraph::opset7::Constant>(transpose_order.get_node_shared_ptr());
+    if (!const_with_order_values)
+        return false;
+
+    const int64_t* data = const_with_order_values->get_data_ptr<int64_t>();
+    if (!data)
+        return false;
+
+    for (size_t i = 0; i < order.size(); i++) {
+        if (order[i] != data[i])
+            return false;
+    }
+
+    return true;
+}
+
+static bool VerifyBias(std::shared_ptr<ngraph::opset7::Add> bias, const size_t& filter_count) {
+    auto add_const = std::dynamic_pointer_cast<ngraph::op::Constant>(bias->input_value(1).get_node_shared_ptr());
```

Comment:
Add is commutative operation so you need to check both inputs.

Reply:
Fill be fixed.